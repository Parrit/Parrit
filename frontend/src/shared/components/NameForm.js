"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var _this = this;
Object.defineProperty(exports, "__esModule", { value: true });
exports.NameForm = void 0;
var react_1 = __importStar(require("react"));
var NameForm = function (props) {
    var _a = react_1.useState(""), name = _a[0], setName = _a[1];
    var handleChange = function (event) {
        setName(event.target.value);
    };
    var submit = function (e) {
        e.preventDefault();
        props.confirmFunction(name);
    };
    var inputClasses = "form-control";
    inputClasses += props.errorMessage ? " error" : "";
    return (react_1.default.createElement("form", { onSubmit: submit.bind(_this) },
        react_1.default.createElement("div", { className: "form-header" },
            react_1.default.createElement("h2", { className: "form-title" }, props.formTitle),
            react_1.default.createElement("div", { className: "form-cancel", onClick: props.cancelFunction })),
        react_1.default.createElement("div", { className: "error-message" }, props.errorMessage),
        react_1.default.createElement("input", { className: inputClasses, autoFocus: true, type: "text", placeholder: "Name", value: name, onChange: handleChange.bind(_this) }),
        react_1.default.createElement("div", { className: "buttons" },
            react_1.default.createElement("button", { type: "submit", className: "button-blue" }, "OK"),
            react_1.default.createElement("button", { type: "button", onClick: props.cancelFunction, className: "button-red" }, "Cancel"))));
};
exports.NameForm = NameForm;
exports.default = exports.NameForm;
