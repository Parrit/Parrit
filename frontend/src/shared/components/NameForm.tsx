import React, { useState } from "react";
import classNames from "classnames";

interface Props {
  formTitle: string;
  confirmFunction: (name: string) => void;
  cancelFunction: VoidFunction;
  errorMessage?: string;
}

export const NameForm: React.FC<Props> = (props) => {
  const [name, setName] = useState("");

  const inputClasses = classNames({
      "form-control": true,
      "error": props.errorMessage != undefined
  })

  const onKeyDownHandler = (event: any) => {
      if (event.key === 'Enter') {
        props.confirmFunction(name);
      }
  }

  return (
    <form onSubmit={(e) => {e.preventDefault()}}>
      <div className="form-header">
        <h2 className="form-title">{props.formTitle}</h2>
        <div className="form-cancel" onClick={props.cancelFunction} />
      </div>

      <div className="error-message">{props.errorMessage}</div>

      <input
        className={inputClasses}
        autoFocus
        type="text"
        placeholder="Name"
        value={name}
        onChange={(event) => setName(event.target.value)}
        onKeyDown={onKeyDownHandler}
      />

      <div className="buttons">
        <button type="button" className="button-blue" onClick={() => props.confirmFunction(name)}>
          OK
        </button>
        <button
          type="button"
          onClick={props.cancelFunction}
          className="button-red"
        >
          Cancel
        </button>
      </div>
    </form>
  );
};

export default NameForm;
