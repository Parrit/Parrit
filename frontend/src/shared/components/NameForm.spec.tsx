import * as React from 'react'
import * as renderer from 'react-test-renderer';
import NameForm from "./NameForm";
import {render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe('<NameForm/>', () => {
    let props;

    beforeEach(() => {
        props = {
            formTitle: 'Form Title',
            confirmFunction: jasmine.createSpy('confirmFunctionSpy'),
            cancelFunction: jasmine.createSpy('cancelFunctionSpy'),
            errorMessage: 'ERROR'
        }
    })

    it('should render correctly', () => {
        const tree = renderer.create(<NameForm {...props}/>)
        expect(tree).toMatchSnapshot();
    })

    describe('submit', () => {
        it('should call the confirm function with the name entered', () => {
            render(<NameForm {...props} />)
            userEvent.type(screen.getByPlaceholderText("Name"), 'stuff');
            userEvent.click(screen.getByText('OK'));
            expect(props.confirmFunction).toHaveBeenCalledWith('stuff')
        })

        it('should call the confirm function with the name entered', () => {
            render(<NameForm {...props} />)
            let input = screen.getByPlaceholderText("Name");
            userEvent.type(input, 'stuff');
            userEvent.type(input, '{enter}');
            expect(props.confirmFunction).toHaveBeenCalledWith('stuff')
        })

        it('should call the cancel function without saving when user click cancel', () => {
            render(<NameForm {...props} />)
            userEvent.type(screen.getByPlaceholderText("Name"), 'stuff');
            userEvent.click(screen.getByText('Cancel'));
            expect(props.confirmFunction).not.toHaveBeenCalled();
            expect(props.cancelFunction).toHaveBeenCalled();
        })
    })
})