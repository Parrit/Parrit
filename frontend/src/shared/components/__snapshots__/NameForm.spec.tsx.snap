// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NameForm/> should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-header"
  >
    <h2
      className="form-title"
    >
      Form Title
    </h2>
    <div
      className="form-cancel"
      onClick={[Function]}
    />
  </div>
  <div
    className="error-message"
  >
    ERROR
  </div>
  <input
    autoFocus={true}
    className="form-control error"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <div
    className="buttons"
  >
    <button
      className="button-blue"
      onClick={[Function]}
      type="button"
    >
      OK
    </button>
    <button
      className="button-red"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</form>
`;
